### ISSUE TITLE
UNNEST aliases in not parsing correctly

### ISSUE THREAD

**Issue by anton-pavlov-simplisafe** [sqlfluff/sqlfluff#7159]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Linting the example query for Athena or Trino dialect (likely others the might use the UNNEST functionality):
```
with dataset as (
    select
        'engineering' as department
        , array['Sharon', 'John', 'Bob', 'Sally'] as users
)
select
    dataset.department
    , t.names
from dataset
cross join unnest(dataset.users) as t(names)
```
the aliases for the unnest output table `t` and column name `names` do not parse correctly.

### Expected Behaviour

The alias for table `t` and its column `names` should be recorgnized as aliases and the query pass all associated linting checks.

### Observed Behaviour

Currently, this query will trigger `AL05: Remove unused table aliases` and remove the aliasing ` as t(names)` altogether when running `sqlfluff fix`.  It is necessary to have the alias in this form because both the table `t` and column `names` are otherwise unnamed.  Also note that we can have multiple columns in the UNNEST output, so the fix should also allow for multiple columns in the aliasing ex.`UNNEST(users) as t(first_name, last_name)`

### How to reproduce

Execute `sqlfluff -fix` on a sql file containing the above example query.

### Dialect

Athena and Trino, although I have seen other dialects share this functionality as well.

### Version

3.4.0, same results after upgrading to 3.4.2 as well.

### Configuration

```
[sqlfluff]

# Set the style for the linter
dialect = athena

# Set the templater to replace {*} with values 
templater = jinja

# Remove linter file size limit
large_file_skip_byte_limit=0

# The following rules are not used when running sqlfluff
# LT08: layout.cte_newline (Blank line expected but not found after CTE closing bracket)
# ST06: structure.column_order (Select wildcards then simple targets before calculations and aggregates)
# ST07: structure.using (Prefer specifying join keys instead of using USING)
# RF01: references.from (References must be present in FROM clause - incompatible with accessing values in nested structs)
# RF04: references.keywords (Keywords should not be used as identifiers)
# RF06: references.quoting (Unnecessary quoted identifier)
exclude_rules = LT08, ST06, ST07, RF01, RF04, RF06
max_line_length = 120 

# This will allow undefined placeholders (usually temp table references) to be replaced with the placeholder name
ignore = templating

[sqlfluff:rules:aliasing.table]
# Aliasing preference for tables
aliasing = explicit

[sqlfluff:rules:aliasing.column]
# Aliasing preference for columns
aliasing = explicit

[sqlfluff:rules:aliasing.length]
# Minimum length of an alias
min_alias_length = 3

[sqlfluff:rules:ambiguous.column_references]
# Address explicitly column names in Group By/ Order By
group_by_and_order_by_style = explicit

[sqlfluff:rules:ambiguous.join]
# Specify types of join clauses
fully_qualify_join_types = inner

[sqlfluff:rules:capitalisation.keywords]
# Lowercase SQL keywords
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
# Lowercase all unquoted identifiers
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
# Lowercase all function names
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
# Lowercase all addressed datatypes
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
# Lowercase all null/true/false literals
capitalisation_policy = lower

[sqlfluff:rules:convention.blocked_words]
# Detect these words in PR
blocked_words = datalake_playground, datalake_pii_playground

[sqlfluff:rules:convention.not_equal]
# The style for using not equal to operator
preferred_not_equal_style = c_style

[sqlfluff:rules:layout.long_lines]
# Line length
ignore_comment_lines = True
ignore_comment_clauses = True

[sqlfluff:indentation]
# No tabs --- Use four spaces for indentation
indent_unit = space
tab_space_size = 4

# ON condition and subsequent columns to be indented
indented_using_on = True
indented_on_contents = False
allow_implicit_indents = True

[sqlfluff:rules:layout.spacing]
no_trailing_whitespace = True

[sqlfluff:rules:layout.select_targets]
wildcard_policy = multiple

[sqlfluff:rules:structure.subquery]
# Forbid subqueries in from and join clauses
forbid_subquery_in = both

[sqlfluff:layout:type:comma]
# Place commas at the beginning of a line
line_position = leading

[sqlfluff:layout:type:where_clause]
keyword_line_position = alone
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
I am unable to reproduce this issue. Do you have any additional logs or information that would help me reproduce this issue?

**Comment by anton-pavlov-simplisafe** [comment]
@keraion I apologize, I pasted a slightly different version of the query.  I updated the query in the original description.  The issue happens when `names` in the select statement is unqualified.  sqlfluff will remove the alias entirely even though the column alias is what is referenced.
`sqlfluff fix` output:
```
==== finding fixable violations ====
== [al05-test.sql] FAIL                                                                                                                                                                                                                                                                             
L:  10 | P:  37 | AL05 | Alias 't' is never used in SELECT statement.
                       | [aliasing.unused]
L:  10 | P:  46 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
== [al05-test.sql] FIXED
2 fixable linting violations found
  [1 unfixable linting violations found]
```

**Comment by anton-pavlov-simplisafe** [comment]
To expand on this, the issue is further complicated when the unnested column is a struct type, and in the SELECT statement, a nested property is accessed, like in the simple example below:
```
with dataset as (
    select
        'engineering' as department
        , cast(
            array[row('Bob', 38), row('John', 29)] as array<row (name varchar, age integer)>
        ) as users
)
select
    dst.department
    , t.user.name
from dataset as dst
cross join unnest(dst.users) as t(user)
```
it looks like what starts happening is that sqlfluff doesn't parse aliasing properly in the select columns when accessing the nested properties of the struct.  This may be a separate issue, but it is closely related because now AL05 will remove the the `t(user)` alias even when it is explicitly referenced.

**Comment by peterbud** [comment]
I'm not sure I understand the original issue.

When take the original sql:
```sql
with dataset as (
    select
        'engineering' as department
        , array['Sharon', 'John', 'Bob', 'Sally'] as users
)
select
    dataset.department
    , names
from dataset
cross join unnest(dataset.users) as t(names)
```

and run with the latest sqlfluff:

```bash
sqlfluff lint test.sql -d athena

== [test.sql] FAIL                                                                                                       
L:   4 | P:  54 | RF04 | Keywords should not be used as identifiers.
                       | [references.keywords]
L:   6 | P:   1 | LT08 | Blank line expected but not found after CTE closing
                       | bracket. [layout.cte_newline]
L:   8 | P:   7 | RF02 | Unqualified reference 'names' found in select with more
                       | than one referenced table/view.
                       | [references.qualification]
L:  10 | P:  37 | AL05 | Alias 't' is never used in SELECT statement.
                       | [aliasing.unused]
L:  10 | P:  38 | LT01 | Expected single whitespace between naked identifier and
                       | start bracket '('. [layout.spacing]
L:  10 | P:  45 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
All Finished 📜 🎉!
```

As you see:
* there is no `AL05: Remove unused table aliases`
* however it correctly calls out that `names` does not use 't' so the `t` alias is not used

If you change the SQL and use the 't.names' qualified reference, it lints properly:

```bash
sqlfluff fix test.sql -d athena

== [test.sql] FAIL                                                                                                       
L:   6 | P:   1 | LT08 | Blank line expected but not found after CTE closing
                       | bracket. [layout.cte_newline]
L:  10 | P:  38 | LT01 | Expected single whitespace between naked identifier and
                       | start bracket '('. [layout.spacing]
L:  10 | P:  45 | LT12 | Files must end with a single trailing newline.
                       | [layout.end_of_file]
== [test.sql] FIXED
3 fixable linting violations found
  [1 unfixable linting violations found]
```

And this is the result:
```sql
with dataset as (
    select
        'engineering' as department
        , array['Sharon', 'John', 'Bob', 'Sally'] as users
)

select
    dataset.department
    , t.names
from dataset
cross join unnest(dataset.users) as t (names)
```

Could you clarify what would be the expected output with 'names' and with 't.names'?

**Comment by peterbud** [comment]
closing due to lack of response
